File: test_datetime.lua
local ffi = require "ffi"

ffi.cdef [[
    const char *dt_time(int argc, ...);
    const char *dt_datetime(int argc, ...);
    const char *dt_date(int argc, ...);
    int64_t dt_unixepoch(int argc, ...)
    double dt_julianday(int argc, ...);
    const char *dt_timediff(int argc, ...);
    const char *dt_strftime(const char *zFmt, int argc, ...);
]]

local S = ffi.load("datetime.so")

print(ffi.string(S.dt_datetime(2,"now","+1 days","localtime")))
print(ffi.string(S.dt_date(2,"now","+1 days","localtime")))
print(ffi.string(S.dt_time(2,"now","+1 days","localtime")))
print(S.dt_julianday(1,"2013-01-01 00:30"))
print(S.dt_unixepoch(2,"now","+1 days","localtime"))
print(ffi.string(S.dt_timediff(2,"now","2013-01-01")))
print(ffi.string(S.dt_strftime("小时 %H",3,"now","localtime","+2 hours")))

File: test_datetime.c
#include "datetime.h"
#include <stdio.h>
#include <time.h>
#include <assert.h>

int main()
{
    assert(dt_unixepoch(1,"now") == time(NULL) && "unixepoch invalid");
	assert(dt_julianday(1, "2000-01-01") == 2451544.5);
	assert(dt_julianday(1, "1970-01-01") == 2440587.5);
	assert(dt_julianday(1, "1910-04-20") == 2418781.5);
	assert(dt_julianday(1, "abc") == 0.0);
	return 0;
}